runtime/sema/type.go

```diff
@@ -6448,6 +6448,25 @@ func (t *AddressType) initializeMemberResolvers() {
 	})
 }
 
+func IsPrimitiveOrContainerOfPrimitive(ty Type) bool {
+	if ty.IsPrimitiveType() {
+		return true
+	}
+
+	switch ty := ty.(type) {
+	case *VariableSizedType:
+		return IsPrimitiveOrContainerOfPrimitive(ty.Type)
+
+	case *ConstantSizedType:
+		return IsPrimitiveOrContainerOfPrimitive(ty.Type)
+
+	case *DictionaryType:
+		return IsPrimitiveOrContainerOfPrimitive(ty.ValueType)
+	}
+
+	return false
```

Comment:
Just a nit-pick: calling for `IsPrimitiveType` for arrays and dictionaries is guranteed to return false.

```suggestion
	switch ty := ty.(type) {
	case *VariableSizedType:
		return IsPrimitiveOrContainerOfPrimitive(ty.Type)

	case *ConstantSizedType:
		return IsPrimitiveOrContainerOfPrimitive(ty.Type)

	case *DictionaryType:
		return IsPrimitiveOrContainerOfPrimitive(ty.ValueType)
		
	default:
	  	return ty.IsPrimitiveType()
	}
```